### ISSUE TITLE
Team Libraries

### ISSUE THREAD

**Issue by knolleary** [flowforge/flowforge#237]
### Description

Node-RED has the concept of a library - a place to import and export flows/node configs to.

By default, there is a single library available only to a particular NR instance. But it also supports plugging in additional library sources.

This Epic introduces the concept of a Team Library.

**This feature will only be available on FlowForge Premium, not FF Open Source**

A user will be able to save flows in one project to their Team Library and then be able to import them into any of the other projects in the Team.

This is all done via the Import/Export dialogs within the Node-RED Editor:

![Image](https://user-images.githubusercontent.com/51083/211620094-d0663bdf-3ffe-4e17-9a05-3d9b986f0719.png)

---

### Technical Details

(I'll move this content over the appropriate PR once it is ready to be published - just capturing it now whilst all fresh in my head)

This is implemented as a custom [Library Store Plugin](https://nodered.org/docs/api/library/) that is setup by the FlowForge Launcher responsible for running Node-RED.

### API Routes

A pair of new API end-points are added alongside the other project storage apis:

 - `POST /storage/:projectId/shared-library/:libraryId/:type`
 - `GET /storage/:projectId/shared-library/:libraryId/:type`

For the initial version, the `:libraryId` parameter must be the Project's Team Hashid, but in the future, could be used to support multiple team libraries (something that is out-of-scope for today).

### Database Updates

The library entries are stored in a new Database table - `StorageSharedLibrary`. This is very similar in structure to the existing `StorageLibrary` table used by the 'local' library in Node-RED. A separate table was needed as `StorageLibrary` uses the Project as a primary key - whereas we want the Team as the primary key for the Shared Library.

### Telemetry Updates

Adds two new metrics to the telemetry:

 - `platform.counts.libraryEntries` - number of entries in the 'local' library database table
 - `platform.counts.sharedLibraryEntries` - number of entries in the new shared library database table

Note that if running without an EE license, the latter metric will always be reported as 0.

**Comment by hardillb** [comment]
Can we move this to verified now?

**Comment by knolleary** [comment]
Verified in staging.
